warp10/src/main/java/io/warp10/standalone/InMemoryChunkSet.java

```diff
@@ -233,6 +238,94 @@ public GTSEncoder fetchEncoder(long now, long timespan) throws IOException {
     
     GTSEncoder encoder = new GTSEncoder(0L);
     
+    //
+    // Fetch the postBoundary
+    //
+    
+    if (postBoundary > 0) {
+      // We store sequence / ts / lat / lon / elev / value
+      // 'sequence' is the order in which the datapoints were inserted, used
+      // to order them when their timestamp is equal
+      PriorityQueue<Object[]> boundary = new PriorityQueue<Object[]>(new Comparator<Object[]>() {
+        @Override
+        public int compare(Object[] o1, Object[] o2) {
+          int comp = Long.compare((long) o1[1], (long) o2[1]);
+          // Timestamps are equal, compare the sequence #
+          if (0 == comp) {
+            comp = Long.compare((long) o1[0], (long) o1[1]);
+          }
+          return comp;
+        }
+      });
+
+      // Now populate the queue from the current chunk, walking back in the ring until we have
+      // postBoundary elements or we have exhausted the available data
+      
+      long seq = 0;
+      
+      for (int i = 0; i < this.chunkcount; i++) {
+        int chunk = (nowchunk + i) % this.chunkcount;
+        
+        GTSDecoder chunkDecoder = null;
+        
+        synchronized(this.chunks) {
+          // Ignore a given chunk if it is before 'now'
+          if (this.chunkends[chunk] - this.chunklen < now) {
```

Comment:
Not sure if I get this right but could we have `this.chunkends[chunk] - this.chunklen < now < this.chunkends[chunk]` so that this chunk contains data for the postBoundary?
In this case, shouldn't the test be `this.chunkends[chunk] < now`?

Reply:
Actually this.chunkends[chunk] <= now and nullity of chunks[chunk]